src/vorta/views/partials/password_input.py

```diff
@@ -0,0 +1,146 @@
+from PyQt5.Qt import QObject
+from PyQt5.QtWidgets import QAction, QFormLayout, QLabel, QLineEdit, QWidget
+from vorta.i18n import translate
+from vorta.views.utils import get_colored_icon
+
+
+class PasswordLineEdit(QLineEdit):
+    def __init__(self, *, parent=None, show_visibility_button=True):
+        super().__init__(parent)
+
+        self._show_visibility_button = show_visibility_button
+        self._error_state = False
+        self._visible = False
+
+        self.setEchoMode(QLineEdit.Password)
+
+        if self._show_visibility_button:
+            self.showHideAction = QAction(self.tr("Show password"), self)
+            self.showHideAction.setCheckable(True)
+            self.showHideAction.toggled.connect(self.toggle_visibility)
+            self.showHideAction.setIcon(get_colored_icon("eye"))
+            self.addAction(self.showHideAction, QLineEdit.TrailingPosition)
+
+    def get_password(self):
+        """Return password text"""
+        return self.text()
+
+    @property
+    def visible(self):
+        """Return password visibility"""
+        return self._visible
+
+    @visible.setter
+    def visible(self, value):
+        """Set password visibility"""
+        if not isinstance(value, bool):
+            raise TypeError("visible must be a boolean value")
+        self._visible = value
+        self.setEchoMode(QLineEdit.Normal if self._visible else QLineEdit.Password)
+
+        if self._show_visibility_button:
+            self.showHideAction.setIcon(get_colored_icon("eye"))
+        else:
+            self.showHideAction.setIcon(get_colored_icon("eye-slash"))
```

Comment:
This is going in the right direction. However you made a logical error. This still fails if `_show_visibility_button` is false.

Reply:
My bad. I mistook `_show_visibility_button` for `_visible`.